crates/ruff_linter/src/rules/pycodestyle/rules/type_comparison.rs

```diff
@@ -53,38 +57,84 @@ pub(crate) fn type_comparison(checker: &mut Checker, compare: &ast::ExprCompare)
+            _ => continue,
+        };
+
+        let left_argument = left_arguments.args.first().unwrap();
+
         // Right-hand side must be, e.g., `type(1)` or `int`.
         match right {
             Expr::Call(ast::ExprCall {
-                func, arguments, ..
+                func: right_func,
+                arguments: right_arguments,
+                ..
             }) => {
                 // Ex) `type(obj) is type(1)`
-                let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() else {
+                let Expr::Name(ast::ExprName { id, .. }) = right_func.as_ref() else {
                     continue;
                 };
 
                 if id == "type" && checker.semantic().is_builtin("type") {
+                    let right_argument = right_arguments.args.first();
+
                     // Allow comparison for types which are not obvious.
-                    if arguments
-                        .args
-                        .first()
-                        .is_some_and(|arg| !arg.is_name_expr() && !is_const_none(arg))
+                    if right_argument.is_some_and(|arg| !arg.is_name_expr() && !is_const_none(arg))
                     {
-                        checker
-                            .diagnostics
-                            .push(Diagnostic::new(TypeComparison, compare.range()));
+                        // find the type of argument if it resolves into a builtin type
+                        let right_side: String = match right_argument.unwrap() {
+                            Expr::Constant(ast::ExprConstant { value, .. }) => match value {
+                                ast::Constant::Str(..) => "str".to_string(),
+                                ast::Constant::Bytes(..) => "bytes".to_string(),
+                                ast::Constant::Int(..) => "int".to_string(),
+                                ast::Constant::Float(..) => "float".to_string(),
+                                ast::Constant::Complex { .. } => "complex".to_string(),
+                                ast::Constant::Bool(..) => "bool".to_string(),
+                                _ => continue,
+                            },
+                            Expr::FString(ast::ExprFString { .. }) => "str".to_string(),
+                            Expr::Tuple(ast::ExprTuple { .. }) => "tuple".to_string(),
+                            Expr::List(ast::ExprList { .. }) => "list".to_string(),
+                            Expr::Set(ast::ExprSet { .. }) => "set".to_string(),
+                            Expr::Dict(ast::ExprDict { .. }) => "dict".to_string(),
+                            Expr::DictComp(_) => "dict".to_string(),
+                            Expr::BoolOp(_) => "bool".to_string(),
+                            Expr::ListComp(_) => "list".to_string(),
+                            Expr::SetComp(_) => "set".to_string(),
+                            _ => checker.generator().expr(right),
+                        };
```

Comment:
is there a convenience method for this kind of thing?

Reply:
Hm there's the more limited https://github.com/astral-sh/ruff/blob/5849a752235f5384bfef733e64dbfc6a01e30be9/crates/ruff_linter/src/rules/ruff/rules/invalid_index_type.rs#L206 

I don't think we've made a broader mapping yet